### PR TITLE
Added ability to use a separate pin as well as the current Z probe

### PR DESCRIPTION
**Pull request by SJ-Innovation**
While using the connection to the Z probe is useful 99% of the time, I can think of a few times when youd want to use the nozzle probe separately to the z_min endstop. Thats what I've added here. 

Instances include, when using z_min as a lower limit safety switch, and when using bed piezos/sensors for levelling

### CODE REVIEW

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -535,6 +535,15 @@
...
 #if ENABLED(CALIBRATION_GCODE)
+
+  #ifndef BACKLASH_COMPENSATION_PIN
+    #define BACKLASH_COMPENSATION_PIN -1 // Override in pins.h or set to -1 to use your Z endstop
+  #endif
+
+  #define BACKLASH_COMPENSATION_PIN_INVERTING false // set to true to invert the backlash compensation nozzle probe
+  //#define PULLDOWN_BACKLASH_COMPENSATION
+  #define PULLUP_BACKLASH_COMPENSATION
+
```
Comment (marcio-ao) [inline Marlin/Configuration_adv.h]: Should maybe BACKLASH_COMPENSATION_PIN_INVERTING and PULLUP_BACKLASH_COMPENSATION be inside the #if def? I assume you only want to use these if BACKLASH_COMPENSATION_PIN is set.

Also, maybe PULLUP_BACKLASH_COMPENSATION should be BACKLASH_COMPENSATION_PIN_PULLUP for consistency.

Comment (marcio-ao) [reply Marlin/Configuration_adv.h]: Also, why are you calling it BACKLASH_COMPENSATION? The calibration routine does much more than just the backlash measurements. It also does nozzle offsets. How about:

- CALIBRATION_CUBE_PROBE_PIN
- CALIBRATION_CUBE_PROBE_PIN_INVERTING
- CALIBRATION_CUBE_PROBE_PIN_PULLUP

**Review by marcio-ao — COMMENTED** [review]
A few suggestions.

File: Marlin/Configuration_adv.h (hunk 1 above)
Comment (SJ-Innovation) [reply Marlin/Configuration_adv.h]: Taken on board, making edits as desired.

Comment (marcio-ao) [reply Marlin/Configuration_adv.h]: Or, maybe just:
- CALIBRATION_PROBE_PIN
- CALIBRATION_PROBE_PIN_INVERTING
- CALIBRATION_PROBE_PIN_PULLUP

**Comment by SJ-Innovation** [comment]
Edits made, dont honestly know why I called them that, seems a bit short sighted of me. Thanks for raising it!

**Pull request merged**